File: crop-me/src/main/java/com/mikelau/croperino/CameraDialog.java
package com.mikelau.croperino;

import android.content.Context;
import android.content.DialogInterface;
import android.content.DialogInterface.OnCancelListener;
import android.os.Build;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

public class CameraDialog {

    private static AlertDialog.Builder sBuilder;

    public static void getConfirmDialog(Context mContext, String mTitle, String mMessage, String mPositive, String mNeutral, String mNegative,
                                        int color, boolean mIsCancelable, final AlertInterface.WithNeutral mTarget) {
        LayoutInflater i = LayoutInflater.from(mContext);
        View v = i.inflate(R.layout.dialog_camera, null);

        if (Build.VERSION.SDK_INT >= 17) {
            sBuilder = new AlertDialog.Builder(mContext, R.style.MyAlertDialogStyle);
        } else {
            sBuilder = new AlertDialog.Builder(mContext);
        }

        sBuilder.setView(v).setCancelable(false).setPositiveButton(mPositive, new DialogInterface.OnClickListener() {
            public void onClick(DialogInterface dialog, int id) {
                mTarget.PositiveMethod(dialog, id);
            }
        }).setNeutralButton(mNeutral, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int id) {
                mTarget.NeutralMethod(dialog, id);
            }
        }).setNegativeButton(mNegative, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int id) {
                mTarget.NegativeMethod(dialog, id);
            }
        });

        TextView tvTitle = v.findViewById(R.id.tv_dialog_title);
        TextView tvMessage = v.findViewById(R.id.tv_dialog_message);

        tvTitle.setBackgroundColor(color);
        tvTitle.setText(mTitle);
        tvMessage.setText(mMessage);

        final AlertDialog alert = sBuilder.create();
        alert.setCancelable(mIsCancelable);
        alert.setCanceledOnTouchOutside(true);
        alert.show();
        if (mIsCancelable) {
            alert.setOnCancelListener(new OnCancelListener() {
                @Override
                public void onCancel(DialogInterface arg0) {
                    mTarget.NegativeMethod(alert, 0);
                }
            });
        }
    }
}

File: croperino/src/main/java/com/mikelau/croperino/CameraDialog.kt
package com.mikelau.croperino

import android.content.Context
import android.view.LayoutInflater
import android.widget.TextView
import androidx.appcompat.app.AlertDialog

object CameraDialog {
    private var sBuilder: AlertDialog.Builder? = null

    fun getConfirmDialog(
        context: Context, title: String, message: String,
        positive: String, neutral: String, negative: String, color: Int,
        isCancelable: Boolean, target: WithNeutral
    ) {
        val i = LayoutInflater.from(context)
        val v = i.inflate(R.layout.dialog_camera, null)
        sBuilder = AlertDialog.Builder(context, R.style.MyAlertDialogStyle)
        sBuilder?.setView(v)?.setCancelable(false)
            ?.setPositiveButton(positive) { dialog, id -> target.PositiveMethod(dialog, id) }
            ?.setNeutralButton(neutral) { dialog, id -> target.NeutralMethod(dialog, id) }
            ?.setNegativeButton(negative) { dialog, id -> target.NegativeMethod(dialog, id) }
        val tvTitle = v.findViewById<TextView>(R.id.tv_dialog_title)
        val tvMessage = v.findViewById<TextView>(R.id.tv_dialog_message)
        tvTitle.setBackgroundColor(color)
        tvTitle.text = title
        tvMessage.text = message
        val alert = sBuilder?.create()
        alert?.setCancelable(isCancelable)
        alert?.setCanceledOnTouchOutside(true)
        alert?.show()
        if (isCancelable) {
            alert?.setOnCancelListener { target.NegativeMethod(alert, 0) }
        }
    }
}
